src/main/java/duke/FileReadWrite.java

```diff
@@ -0,0 +1,139 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+/**
+ * Class that handles writing and reading from the duke.txt file
+ */
+public abstract class FileReadWrite {
+    private static final String FILE_PATH = Paths.get(Paths.get(System.getProperty("user.dir")).toString(),
+            "data/duke.txt").toString();
+
+
+    /**
+     * Returns lateral location of the specified position.
+     * If the position is unset, NaN is returned.
+     *
+     * @return Task List that is read and interpreted from the duke.txt file
+     * @throws IOException  If the user input is interrupted.
+     */
+    public static TaskList readFile() throws IOException {
+        File file = new File(FILE_PATH);
+        TaskList taskList = new TaskList();
+        if (!file.exists()) {
+            file.mkdir();
+        }
+        Scanner scanner = new Scanner(file);
```

Comment:
It seems that the `Scanner scanner` was never closed and can cause a resource leak. Maybe you can close the scanner right before you `return taskList`?

This occurs in the following as well:
 ```
    public static void writeMark(int i, Task t) throws IOException {
        File file = new File(FILE_PATH);
        Scanner scanner = new Scanner(file);
 ```
 ```
     public static void writeUnmark(int i, Task t) throws IOException {
        File file = new File(FILE_PATH);
        Scanner scanner = new Scanner(file);
 ```
 ```
     public static void writeDelete(int i) throws IOException {
        File file = new File(FILE_PATH);
        Scanner scanner = new Scanner(file);
 ```